system/HTTP/CURLRequest.php

```diff
@@ -469,7 +469,7 @@ protected function applyMethod(string $method, array $curlOptions): array
      */
     protected function applyBody(array $curlOptions = []): array
     {
-        if (! empty($this->body)) {
+        if ($this->body !== '' && $this->body !== null) {
```

Comment:
can we just cast to string then check against empty string?

Reply:
You mean like this?

```diff
$ git diff system/HTTP/CURLRequest.php
diff --git a/system/HTTP/CURLRequest.php b/system/HTTP/CURLRequest.php
index c78f088ff7..40e197abb5 100644
--- a/system/HTTP/CURLRequest.php
+++ b/system/HTTP/CURLRequest.php
@@ -469,8 +469,10 @@ class CURLRequest extends OutgoingRequest        
      */
     protected function applyBody(array $curlOptions = []): array
     {
-        if ($this->body !== '' && $this->body !== null) {
-            $curlOptions[CURLOPT_POSTFIELDS] = (string) $this->getBody();
+        $requestBody = (string) $this->getBody();
+
+        if ($requestBody !== '') {
+            $curlOptions[CURLOPT_POSTFIELDS] = $requestBody;
         }

         return $curlOptions;
```